radix-engine-tests/tests/events.rs

```diff
@@ -0,0 +1,92 @@
+use radix_engine::errors::{KernelError, RuntimeError};
+use radix_engine::transaction::TransactionReceipt;
+use radix_engine::types::*;
+use radix_engine_interface::api::LockFlags;
+use scrypto_unit::*;
+use transaction::builder::ManifestBuilder;
+use transaction::data::manifest_args;
+
+fn test_event_store_locking_from_scrypto(lock_flags: LockFlags) -> TransactionReceipt {
+    let mut test_runner = TestRunner::builder().without_trace().build();
+    let package_address = test_runner.compile_and_publish("./tests/blueprints/events");
+
+    let receipt = {
+        let args = manifest_args!(lock_flags.bits());
+        let manifest = ManifestBuilder::new()
+            .call_function(
+                package_address,
+                "EventStoreVisibility",
+                "lock_event_store",
+                args,
+            )
+            .build();
+        test_runner.execute_manifest_ignoring_fee(manifest, vec![])
+    };
+    receipt
+}
+
+/// Tests that Scrypto code can't lock the event store. This is to ensure that Scrypto code can't
+/// add arbitrary events by manually locking the substate and adding events into it.
```

Comment:
I'd add as a TODO that we don't even expose the EventStore RENode to the client. But this will only be possible after a bit more refactors in the system layer.

Reply:
That would be cool. In that model, how would Scrypto blueprints emit events?